lib/app/features/chat/messages/views/pages/messages_page.dart

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/chat/messages/views/components/chat_date_header_text/chat_date_header_text.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/audio_message/audio_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/emoji_message/emoji_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/money_message/money_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/photo_message/photo_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/poll_message/poll_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/profile_share_message/profile_share_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/text_message/text_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/url_preview_message/url_preview_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/message_types/video_message/video_message.dart';
+import 'package:ion/app/features/chat/messages/views/components/messaging_bottom_bar/messaging_bottom_bar.dart';
+import 'package:ion/app/features/chat/messages/views/components/messaging_header/messaging_header.dart';
+import 'package:ion/app/features/chat/model/money_message_type.dart';
+import 'package:ion/app/services/keyboard/keyboard.dart';
+
+const String hasPrivacyModalShownKey = 'hasPrivacyModalShownKey';
+
+class MessagesPage extends ConsumerWidget {
+  const MessagesPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return GestureDetector(
+      onTap: () => hideKeyboard(context),
+      child: Scaffold(
+        backgroundColor: context.theme.appColors.secondaryBackground,
+        body: SafeArea(
+          minimum: EdgeInsets.only(
+            bottom: MediaQuery.of(context).padding.bottom > 0 ? 17.0.s : 0,
+          ),
+          bottom: false,
+          child: Column(
+            children: [
+              const MessagingHeader(),
+              // const MessagingEmptyView(),
+              Expanded(
+                child: ColoredBox(
+                  color: context.theme.appColors.primaryBackground,
+                  child: ListView.separated(
+                    itemBuilder: (context, index) {
+                      switch (index) {
```

Comment:
Can you refactor this switch condition?

Reply:
it is just for mocking/testing, when we bind the backend, ll be refactored totally. is it ok for you @ice-alcides ?

Reply:
sure, understood